bricks.js

```diff
@@ -1,64 +1,205 @@
+			rect.setAttributeNS(null, 'y', mortar + unitheight * y);
+			rect.setAttributeNS(null, 'height', brickheight);
+			rect.setAttributeNS(null, 'width', brickwidth);
 			rect.setAttributeNS(null, 'fill', 'rgb(192,75,75)');
-			rect.setAttributeNS(null, 'id', 'B' + String.fromCharCode(65 + y) + num.toString());
+			rect.setAttributeNS(null, 'id', 'B' + xcoord.toString() + '-' + ycoord.toString());
 			rect.setAttributeNS(null, 'class', 'brick');
 			document.getElementById("canvas").appendChild(rect);
 		}
 	}
-	for (var y = 0; y < height + 1; y++) {
-		for (var x = 0; x < 2 * width; x++) {
+}
+
+function drawhorizmortar(nColumns, nRows) {
+	for (var y = 0; y < nRows + 1; y++) {
+		for (var x = 0; x < 2 * nColumns; x++) {
 			var rect = document.createElementNS(svgns, 'rect');
-			var num = x + 1;
-			rect.setAttributeNS(null, 'x', 10 + 55 * x);
-			rect.setAttributeNS(null, 'y', 60 * y);
-			rect.setAttributeNS(null, 'height', 10);
-			rect.setAttributeNS(null, 'width', 45);
-			rect.setAttributeNS(null, 'id', 'L' + String.fromCharCode(65 + y) + num.toString());
+			var xcoord = x + 1;
+			var ycoord = y + 1;
+			rect.setAttributeNS(null, 'x', mortar + halfunit * x);
+			rect.setAttributeNS(null, 'y', unitheight * y);
+			rect.setAttributeNS(null, 'height', mortar);
+			rect.setAttributeNS(null, 'width', horizmortarlength);
+			rect.setAttributeNS(null, 'id', 'H' + xcoord.toString() + '-' + ycoord.toString());
 			rect.setAttributeNS(null, 'class', 'unclicked');
 			rect.setAttributeNS(null, 'onclick', 'borderclick(this)');
 			document.getElementById("canvas").appendChild(rect);
 		}
 	}
-	for (var y = 0; y < height; y++) {
-		for (var x = 0; x < (width + 1 - y % 2); x++) {
+}
+
+function drawvertmortar(nColumns, nRows) {
+	for (var y = 0; y < nRows; y++) {
+		for (var x = 0; x < (nColumns + 1 - y % 2); x++) {
 			var rect = document.createElementNS(svgns, 'rect');
-			var num = x + 1;
-			rect.setAttributeNS(null, 'x', 110 * x + 55 * (y % 2));
-			rect.setAttributeNS(null, 'y', 10 + 60 * y);
-			rect.setAttributeNS(null, 'height', 50);
-			rect.setAttributeNS(null, 'width', 10);
-			rect.setAttributeNS(null, 'id', 'T' + String.fromCharCode(65 + y) + num.toString());
+			var xcoord = x + 1;
+			var ycoord = y + 1;
+			rect.setAttributeNS(null, 'x', unitlength * x + halfunit * (y % 2));
+			rect.setAttributeNS(null, 'y', mortar + unitheight * y);
+			rect.setAttributeNS(null, 'height', brickheight);
+			rect.setAttributeNS(null, 'width', mortar);
+			rect.setAttributeNS(null, 'id', 'V' + xcoord.toString() + '-' + ycoord.toString());
 			rect.setAttributeNS(null, 'class', 'unclicked');
 			rect.setAttributeNS(null, 'onclick', 'borderclick(this)');
 			document.getElementById("canvas").appendChild(rect);
 		}
 	}
 }
 
-drawpuzzle(5,5);
+function drawxs(nColumns, nRows) {
+	/* drawing xs under horizontal mortar */
+	for (var y = 0; y < nRows + 1; y++) {
+		for (var x = 0; x < 2 * nColumns; x++) {
```

Comment:
let's use a [query selector](https://developer.mozilla.org/en-US/docs/Web/API/Element/querySelectorAll) to solve this problem. You know you want to put an X under every piece of mortar, so it's best to write code that expresses that! I'm thinking something like this:

When you're creating the mortar elements above:

```
rect.setAttribute("type", "mortar");
```

Then you can find every such element like this:

```
const mortars = document.querySelectorAll("[type=mortar]");
mortars.forEach((mortar) => {
  if (mortar.id.charAt(0) === 'H') {
    // code for horizontal x
    // you can use like parseInt(mortar.getAttribute("x")) to get the x coordinate of *this particular mortar*
    // rather than compute what you know it ought to be.
  } else {
    // code for vertical x
  }
});
```

We're using an [attribute selector](https://developer.mozilla.org/en-US/docs/Learn/CSS/Introduction_to_CSS/Attribute_selectors) here, but you can also select by class or id or tag.